autoscale_cloudroast/test_repo/autoscale/fixtures.py

```diff
@@ -25,7 +25,111 @@
 from autoscale.otter_constants import OtterConstants
 
 
+def _make_client(access_data, service_name, region, client_cls, debug_name):
+    """
+    Create a client using the `client_cls`, assuming it takes a url, an
+    auth token, serialize format, and deserialize format.
+
+    Extacts the URL and token from the access_data.
+
+    The autoscale client is special in that it extracts the URL from the
+    config, in some cases.
+    """
+    url = None
+
+    # If not production or staging, always use the configured server
+    # endpoint for autoscale instead of what's in the service catalog.
+    if (service_name == "autoscale" and
+            autoscale_config.environment not in ('production', 'staging')):
+        url = "{0}/{1}".format(
+            autoscale_config.server_endpoint, autoscale_config.tenant_id)
+        print(" ------ Using non-production, non-staging otter --------")
+
+    else:
+        service = access_data.get_service(service_name)
+        if service is not None:
+            url = service.get_endpoint(region).public_url
+
+    if url is not None:
+        return client_cls(url, access_data.token.id_, 'json', 'json')
+    else:
+        print("This account does not support {0}.".format(debug_name))
+
+
+def _set_up_clients():
+    """
+    Read the user creds from the configuration file in and constructs all the
+    service clients.  If it can't authenticate, or it cannot construct the
+    autoscale/server/lbaas clients, then it fails.
+
+    The RCv3 client is not created if the account does not have access to RCv3
+    or if RCv3 configuration parameters are not present or invalid.
+    """
+    user_config = UserConfig()
+    access_data = AuthProvider.get_access_data(endpoint_config, user_config)
+
+    if access_data is None:
+        raise Exception(
+            "Unable to authenticate against identity to get auth token and "
+            "service catalog.")
+
+    autoscale_client = _make_client(
```

Comment:
Can you have different names of client variables that don't match global variable names? This can be `_autoscale_client` to emphasize the localness